components/camel-file-watch/src/main/java/org/apache/camel/component/file/watch/FileWatchEndpoint.java

```diff
@@ -0,0 +1,211 @@
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.component.file.watch.constants.FileEventEnum;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.support.DefaultEndpoint;
+
+/**
+ * The file-watch is used to monitor file events in directory using {@link java.nio.file.WatchService}
+ */
+@UriEndpoint(firstVersion = "3.0.0", scheme = "file-watch", title = "file-watch", syntax = "file-watch:path",
+label = "file", consumerOnly = true)
+public class FileWatchEndpoint extends DefaultEndpoint implements MultipleConsumersSupport {
+    @UriPath(label = "consumer", description = "Path of directory to consume events from.")
+    @Metadata(required = true)
+    private String path;
+
+    @UriParam(label = "consumer",
+    description = "Coma separated list of events to watch. Allowed values are: CREATE, MODIFY, DELETE.",
+    defaultValue = "CREATE,MODIFY,DELETE")
+    private Set<FileEventEnum> events = new HashSet<>(Arrays.asList(FileEventEnum.values()));
+
+    @UriParam(label = "consumer", description = "Auto create directory if does not exists.", defaultValue = "true")
+    private boolean autoCreate = true;
+
+    @UriParam(label = "consumer", description = "Watch recursive in current and child directories (including newly created directories).", defaultValue = "true")
+    private boolean recursive = true;
+
+    @UriParam(label = "consumer",
+    description = "The number of concurrent consumers. Increase this value, if your route is slow to prevent buffering in queue.",
+    defaultValue = "1")
+    private int concurrentConsumers = 1;
+
+    @UriParam(label = "consumer",
+    description = "The number of threads polling WatchService. Increase this value, if you see OVERFLOW messages in log.",
+    defaultValue = "1")
+    private int pollThreads = 1;
+
+    @UriParam(label = "consumer",
+    description = "ANT style pattern to match files. The file is matched against path relative to endpoint path. "
+    + "Pattern must be also relative (not starting with slash)",
+    defaultValue = "**")
+    private String antInclude;
+
+    @UriParam(label = "consumer", description = "Maximum size of queue between WatchService and consumer. Unbounded by default.",
+    defaultValue = "" + Integer.MAX_VALUE)
+    private int queueSize = Integer.MAX_VALUE;
+
+    @UriParam(label = "consumer",
+    description = "Reference to io.methvin.watcher.hashing.FileHasher. "
+    + "This prevents emitting duplicate events on some platforms. "
+    + "For working with large files and if you dont need detect multiple modifications per second per file, "
+    + "use #lastModifiedTimeFileHasher. You can also provide custom implementation in registry.",
+    defaultValue = "#murmur3FFileHasher")
+    private FileHasher fileHasher = FileHasher.DEFAULT_FILE_HASHER;
```

Comment:
You can take some of these options and add on component level too, so you can configure them once, this is better than the registry trick you do above.

Also for example the queueSize so you can set this globally for all file-watcher endpoints.

Then you set these when you create the endpoint from the component on the endpoint instance

Reply:
I have added `concurrentConsumers` `queueSize` `pollThreads` `fileHasher` and `useFileHashing` on the component too.